### DISCUSSION TITLE
Free RoadNodes

### DISCUSSION THREAD

**Discussion by DLList** [INSPIRE-MIF/helpdesk-validator#805] [General]
Hello,

I have a problem with the ETF-Validator that I'd like to solve here.

I am currently busy developing a tool for transforming street data into INSPIRE Transport Network Structures. I have created several INSPIRE Features and tested them with the Validator. I have a problem with free RoadNodes.

**My approach for creating INSPIRE-Data**
Transfer the source data to INSPIRE as it is. This means I get sometimes small geometric differences between RoadNodes and the corresponding RoadLinks in INSPIRE data. These discrepancies sometimes amount to only a few millimetres. 

**Testing the DataSet with the Validator**
The Validator throws in situations of discrepancies the error "no free transport nodes".

**Question**
My question is: How can I solve this problem with free RoadNodes. Maybe should I use for the data the "ConnectivityTolerance"? If so, how to do it?

Best regards,
Dawid

**Comment by josemasensio** [comment]
Dear @DLList,

Thank you for your comment. We are going to begin with a research to provide you with the best solution as soon as possible. Please be patient, this might take some time.

Best regards.

**Comment by josemasensio** [comment]
Dear @DLList,

Please, could you use our [contribution guidelines](https://github.com/inspire-eu-validation/community/blob/master/contribution_guidelines.md) to help us more quickly locate what is causing this issue? 

We will like to have the file you are using, the problem you have and if is possible, a screenshot of the issue. Thanks in advice, this information will be useful for us.

Best regards.

**Comment by josemasensio** [comment]
Dear @DLList,

Please, we will appreciate if you can help us using the  [contribution guidelines](https://github.com/inspire-eu-validation/community/blob/master/contribution_guidelines.md). Otherwise, we could not help you.

Best regards.

**Comment by fabiovin** [comment]
Dear @DLList,
looking at the [related ETS](https://github.com/inspire-eu-validation/ets-repository/blob/master/data-tn/tn-as/ets-tn-as-bsxets.xml) it seems that there is a limitation in the use of a connectivityTolerance parameter (_TestCase: Network connectivity_). Therefore, the related test has been classified as "manual" in the ETS.

@josemasensio, the limitation seems due to the fact that: 

> "the use a connectivityTolerance parameter would require all geometries are first converted to a meter-based CRS, eg ETRS89 LAEA or LCC, which is currently not supported by the geometry library"

Could you verify if the connectivityTolerance parameter can be used in case of the geometries are already in a meter-based CRS?

**Comment by DLList** [comment]
> 
> 
> Dear @DLList,
> 
> Please, could you use our [contribution guidelines](https://github.com/inspire-eu-validation/community/blob/master/contribution_guidelines.md) to help us more quickly locate what is causing this issue?
> 
> We will like to have the file you are using, the problem you have and if is possible, a screenshot of the issue. Thanks in advice, this information will be useful for us.
> 
> Best regards.


Dear @josemasensio,

the problem for me now is more theoretical. Specifically: how to use/set the connectivityTolerance parameter in INSPIRE-XML-File?

Best regards

**Comment by DLList** [comment]
Dear @josemasensio,

In addition to the previous message, I send an example of test data where I would have liked to use the ConnectivityTolerance parameter.

Best regards

[Inspire-A2_NRW.zip](https://github.com/inspire-eu-validation/community/files/3887051/Inspire-A2_NRW.zip)

**Comment by fabiovin** [comment]
Dear @DLList,

there is no way to indicate the ConnectivityTolerance parameter inside the gml dataset.

Please have a look at section _10.2 Ensuring Network Connectivity_ of the Data Specification (page 169), which indicates that: 

> The connectivity tolerance will be supplied as metadata with the transport data. (see Section 7.1.6)

In section 7.1.6 it is indicated how to document connectivity tolerance in the metadata.

**Comment by DLList** [comment]
Dear @fabiovin,

thank you for the answer firstly. I will analyze now this topic.
